src/TinyGsmClientSIM7600.h

```diff
@@ -142,7 +227,163 @@ class TinyGsmSim7600 : public TinyGsmModem<TinyGsmSim7600>,
+      // ^^ The data length which not read in the buffer
+      for (int i = 0; i < len_confirmed; i++) {
+        uint32_t startMillis = millis();
+#ifdef TINY_GSM_USE_HEX
+// (todo) Don't know how to implement this!
+        while (stream.available() < 2 &&
+               (millis() - startMillis < sockets[mux]->_timeout)) {
+          TINY_GSM_YIELD();
+        }
+        char buf[4] = {
+            0,
+        };
+        buf[0] = stream.read();
+        buf[1] = stream.read();
+        char c = strtol(buf, NULL, 16);
+#else
+        while (!at->stream.available() &&
+               (millis() - startMillis < at->sockets[mux]->_timeout)) {
+          TINY_GSM_YIELD();
+        }
+        char c = at->stream.read();
+#endif
+        at->sockets[mux]->rx.put(c);
+      }
+      // DBG("### READ:", len_requested, "from", mux);
+      // sockets[mux]->sock_available = len_confirmed;
+      String await_response = "+CCHRECV: " + String(mux) + ",0";
+      at->waitResponse(await_response.c_str());
+      at->sockets[mux]->sock_available = (uint16_t)modemGetAvailable(mux);
+      return len_confirmed;
+    }
+
+    size_t modemGetAvailable(uint8_t mux) override {
+      size_t result = 0;
+      if (!at->sockets[mux]) return 0;
+      at->sendAT(GF("+CCHRECV?"));
+      if (at->waitResponse(GF(AT_NL "+CCHRECV: ")) != 1) {
+        at->sendAT(GF("+CIPRXGET=4,"), mux);
+        size_t result = 0;
+        if (at->waitResponse(GF("+CIPRXGET:")) == 1) {
+          at->streamSkipUntil(',');  // Skip mode 4
+          at->streamSkipUntil(',');  // Skip mux
+          result = at->streamGetIntBefore('\n');
+          at->waitResponse();
+        }
+        // DBG("### Available:", result, "on", mux);
+        if (!result) {
+          at->sockets[mux]->sock_connected = at->modemGetConnected(mux);
+        }
+        return result;
+      }
+      at->streamSkipUntil(',');  // Skip mode 4
+      if (mux) {
+        at->streamSkipUntil(',');  // Skip mode 4
+        result = at->streamGetIntBefore('\n');
+      } else {
+        result = at->streamGetIntBefore(',');
+      }
+      return result;
+    }
+
+    int connect(const char* host, uint16_t port, int timeout_s) override {
+      stop(15000L);
+      TINY_GSM_YIELD();
+      rx.clear();
+      if (certValidation && at->certificates[mux].length() == 0) {return -1;}
+      sock_connected = at->modemConnect(host, port, mux, sslVersion,
+                                        timeout_s);
+      return sock_connected;
+    }
+
+     void stop(uint32_t maxWaitMs) override {
+      at->sendAT(GF("+CCHCLOSE="), mux);
+      at->waitResponse(5000L);
+
+      if (certificates[mux].length() != 0) {
+        deleteCertificate(certificates[mux].c_str());
+      }
+
+      if (!clientCertificates[mux].length() != 0) {
+        deleteCertificate(clientCertificates[mux].c_str());
+      }
+
+      if (!clientPrivateKeys[mux].length() != 0) {
+        deleteCertificate(clientPrivateKeys[mux].c_str());
+      }
```

Comment:
IMHO, the one adding certificates is also responsible for deleting them, so we don't want to delete them unconditionally here.